src/main/java/dev/ecattez/shahmat/board/move/MovingStrategy.java

```diff
@@ -0,0 +1,17 @@
+package dev.ecattez.shahmat.board.move;
+
+import dev.ecattez.shahmat.board.Board;
+import dev.ecattez.shahmat.board.Piece;
+import dev.ecattez.shahmat.board.Square;
+import dev.ecattez.shahmat.event.BoardEvent;
+
+import java.util.List;
+import java.util.Optional;
+
+public interface MovingStrategy {
+
+    Optional<Movement> getMovement(Piece piece, Square from, Square to, Board board, List<BoardEvent> history);
```

Comment:
If the requested movement is invalid, it will be empty ? Wouldn't it be clearer to have an exception ?

Reply:
I could have a `get` and a `find` to have both Exception and Optional depending on the context